### PR TITLE
Implemented workload runs and NOPE tool support

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Resolves [NETOBSERV-503](https://issues.redhat.com/browse/NETOBSERV-503)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -330,13 +398,99 @@ pipeline {
...
+                        else if (returnCode.toInteger() != 0) {
+                            error('NOPE tool failed :(')
+                        }
+                        else {
+                            println 'Successfully ran NOPE tool :)'
+                        }
+                    }
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply Jenkinsfile:486]: Will address in a separate PR, would you mind making a subtask for this under [NETOBSERV-294](https://issues.redhat.com/browse/NETOBSERV-294)? You can assign it to me

File: scripts/nope.py [hunk 2]
```diff
@@ -292,6 +292,7 @@ def main():
         except Exception as e:
             logging.error(f"Error uploading to Elasticsearch server: {e}\nA local dump to {DATA_DIR}/data_{timestamp}.json will be done instead")
             dump_data_locally(timestamp)
+            sys.exit(2)
```
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply scripts/nope.py:295]: Good question! Normally I just use exit codes 0 and 1, but I wanted a way for the NOPE tool to be able to tell Jenkins that ES upload failed but a local JSON dump succeeded, so we could mark the job as UNSTABLE. I researched exit codes and couldn't find any specific convention on a good number to use for this purpose, so I arbitrarily chose 2. If you know of a more conventional choice I'd be happy to change!

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (memodi) [reply Jenkinsfile:486]: Created this task: https://issues.redhat.com/browse/NETOBSERV-570 for this and https://issues.redhat.com/browse/NETOBSERV-571 to publish to spreadsheets automatically.

File: scripts/nope.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (memodi) [reply scripts/nope.py:295]: Ah, okay. Typically, exit code 1 suits well here for all kinds of generic errors, but if using exit code 2 gives you better handle in Jenkins pipeline, I guess we should be okay, but could you add comment in code on how/why it's used?

[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply scripts/nope.py:295]: Sure, I will add a note. I would have used exit code 1 but we can't here, as we need to uniquely identify this scenario specifically.

**Comment by memodi** [comment]
LGTM for me, I'll wait for @paigerube14 and @Amoghrd to review before adding label.

File: Jenkinsfile [hunk 3]
```diff
@@ -330,13 +398,99 @@ pipeline {
...
+                expression { params.WORKLOAD != 'None' }
+            }
+            steps {
+                script {
+                    if (params.WORKLOAD == 'router-perf') {
+                        jobName = 'scale-ci/e2e-benchmarking-multibranch-pipeline/router-perf'
+                        workloadJob = build job: jobName, parameters: [
+                            string(name: 'BUILD_NUMBER', value: params.FLEXY_BUILD_NUMBER),
+                            string(name: 'JENKINS_AGENT_LABEL', value: params.JENKINS_AGENT_LABEL),
+                            booleanParam(name: 'GEN_CSV', value: false)
```
Comment (paigerube14) [inline Jenkinsfile:412]: did we want to set this to true in the next iteration of this file?

Comment (nathan-weinberg) [reply Jenkinsfile:412]: We're going to do all the spreadsheet-related changes in a followup PR

**Comment by paigerube14** [comment]
LGTM

**Comment by paigerube14** [comment]
@Amoghrd @memodi can you add your final comment to this so it will be auto merged

**Pull request merged**